Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemsBusinessService.cs

```diff
@@ -176,6 +176,21 @@ public async Task<bool> UserHasProblemPermissions(int problemId, string? userId,
             return await this.contestsBusiness.UserHasContestPermissions(problem.ContestId, userId, isUserAdmin);
         }
 
+        public async Task ReevaluateProblemsByOrderBy(int contestId)
+        {
+            var orderByIndex = 0;
+
+            await this.contestsData.GetProblemsById(contestId)
+                                    .OrderBy(p => p.OrderBy)
+                                    .ForEachAsync(p =>
+                                    {
+                                        p.OrderBy = ++orderByIndex;
+                                        this.problemsData.Update(p);
+                                    });
```

Comment:
I'm not sure `ForEachAsync` is the way to go here. We want to update the problems synchronously, as you are updating with a single variable `orderByIndex` which we want to increase for each problem in the given order, not having the value be updated in parallel runs.